test/e2e/storage/csi_mock_volume.go

```diff
@@ -0,0 +1,462 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"regexp"
+	"time"
+
+	"k8s.io/api/core/v1"
+	storage "k8s.io/api/storage/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	volumeutil "k8s.io/kubernetes/pkg/volume/util"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/drivers"
+	"k8s.io/kubernetes/test/e2e/storage/testsuites"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+type cleanupFuncs func()
+
+const (
+	csiNodeLimitUpdateTimeout  = 5 * time.Minute
+	csiPodUnschedulableTimeout = 2 * time.Minute
+)
+
+var _ = utils.SIGDescribe("CSI mock volume", func() {
+	type mockDriverSetup struct {
+		cs           clientset.Interface
+		config       *testsuites.PerTestConfig
+		testCleanups []cleanupFuncs
+		pods         []*v1.Pod
+		pvcs         []*v1.PersistentVolumeClaim
+		sc           map[string]*storage.StorageClass
+		driver       testsuites.TestDriver
+		nodeLabel    map[string]string
+		provisioner  string
+	}
+	var (
+		m               mockDriverSetup
+		attachable      bool
+		deployCRD       bool
```

Comment:
this should be renamed to "deployDriver"

Reply:
renamed to `registerDriver`.